sync/etcdv3/etcd.go

```diff
@@ -47,7 +47,8 @@ type Sync struct {
 }
 
 func (s *Sync) withTimeout() context.Context {
-	ctx, _ := context.WithTimeout(context.Background(), s.timeout)
+	ctx, cancel := context.WithTimeout(context.Background(), s.timeout)
```

Comment:
`ctx, _ := context.WithTimeout(context.Background(), s.timeout)` instead

Reply:
oh, thats `go vet` complaining. `the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak`. I think that "clearing" this problem in such way isn't good.

Reply:
My intention in this PR is not to fix the implementation but only clear linter warnings.
I am going to enforce the state that no more linter warning are added, not fixing existing code. In fact, I tried to fix this implementation several times but whenever I add a call to cancel it causes failing tests. A cause is tat this code is quite complicated.
I asked one the autor of this part for help with this topic but in his opinion there fixing this is a waste of time.
@yudai what dou you think about fixing this code properly?

Reply:
@marcin-ptaszynski if you have some spare time, maybe you can help with this code?

Reply:
@przemyslaw-dobrowolski-cl , Let me take a look at it in spare time. I don't think this PR has big priority.